manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/ConnectorWorker.java

```diff
@@ -48,7 +52,23 @@ protected PanacheRepositoryBase<ConnectorEntity, String> getDao() {
     }
 
     @Override
-    public ConnectorEntity createDependencies(Work work, ConnectorEntity connectorEntity) {
+    protected String getId(JobExecutionContext context) {
+        // The ID of the ManagedResource to process is the child of that stored directly in the JobDetail.
+        JobDataMap data = context.getTrigger().getJobDataMap();
+        String processorId = data.getString(STATE_FIELD_ID);
+        ConnectorEntity connectorEntity = connectorsDAO.findByProcessorId(processorId);
+
+        if (Objects.isNull(connectorEntity)) {
+            //Work has been scheduled but cannot be found. Something (horribly) wrong has happened.
+            throw new IllegalStateException(String.format("Connector for Processor with id '%s' cannot be found in the database.", processorId));
+        }
+
+        return connectorEntity.getId();
+    }
+
+    @Override
+    @ActivateRequestContext
```

Comment:
What is this annotation used for?

Reply:
This is nice... 

`ConnectorWorker` uses `ConnectorsApiClient` which is `@RequestScoped` however Quartz is running `ConnectorWorker` on a thread that does not have a request scope active. This annotation ensures CDI activates a request scope to prevent use of `ConnectorsApiClient` from failing. I went for this solution than changing the scope of `ConenctorsApiClient` as that could have had further reaching consequences.